### PR TITLE
G11 Time

### PR DESCRIPTION
**Pull request by LadyFowls**
Adds the G11 and a heavy ammo type for it for a AMR replacement. Should serve the balence of a weapon that can be used in normal combat without alot of the problems and stigma from the AMR

Aseprite_FItjbJSe5g

### CODE REVIEW

**Comment by dragoslava** [comment]
!review

**Comment by ComradeCorbyn** [comment]
I'm not a spriter, but it just seems too dark for me. I like the color scheme - but maybe make the difference between the gray and the black more prominent? Just looked like one big whole black thing from afar. 

Don't take that quote out of context.

**Comment by NiftyBaby** [comment]
Show in-game screenshot. Thanks.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 1]
```diff
@@ -84,11 +84,21 @@
 	item_cost = 6
 	path = /obj/item/martial_manual
 
+/*
+
+//This is commented out incase somebody wants to easily add it back in
```
Comment (LordFowl) [inline code/datums/uplink/highly visible and dangerous weapons.dm]: No. If someone wants to add it back in they can code it back in. We don't leave commented out code.

File: code/datums/uplink/ammunition.dm [hunk 2]
```diff
@@ -30,7 +30,21 @@
 	item_cost = 4
 	path = /obj/item/ammo_magazine/tommydrum
 
+/*
```
Comment (LordFowl) [inline code/datums/uplink/ammunition.dm]: Delete it if it's not being used.

File: code/game/objects/items/devices/uplink_random_lists.dm [hunk 3]
```diff
@@ -42,8 +42,8 @@ var/datum/uplink_random_selection/default_uplink_selection = new/datum/uplink_ra
 	items += new/datum/uplink_random_item(/datum/uplink_item/item/ammo/mc9mm)
 	items += new/datum/uplink_random_item(/datum/uplink_item/item/visible_weapons/revolver)
 	items += new/datum/uplink_random_item(/datum/uplink_item/item/ammo/a357)
-	items += new/datum/uplink_random_item(/datum/uplink_item/item/visible_weapons/heavysniper, 15, 0)
-	items += new/datum/uplink_random_item(/datum/uplink_item/item/ammo/sniperammo, 15, 0)
+	//items += new/datum/uplink_random_item(/datum/uplink_item/item/visible_weapons/heavysniper, 15, 0)
```
Comment (LordFowl) [inline code/game/objects/items/devices/uplink_random_lists.dm]: Delete.

File: code/modules/projectiles/ammunition/bullets.dm [hunk 4]
```diff
@@ -42,6 +42,19 @@
 	caliber = "trod"
 	projectile_type = /obj/item/projectile/bullet/trod
 
+
+/obj/item/ammo_casing/caseless445m
+	desc = "Caseless ammo fragments."
```
Comment (LordFowl) [inline code/modules/projectiles/ammunition/bullets.dm]: If they are caseless they should not be leaving an ammo casing. There is a defined functionality specifically for caseless ammunition.

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 5]
```diff
@@ -343,6 +343,57 @@
 		list(mode_name="dual coil",	burst=2, move_delay=5, accuracy = list(-2,-3), dispersion = list(20))
 		)
 
+/obj/item/weapon/gun/projectile/automatic/g11
+	name = "G11"
```
Comment (LordFowl) [inline code/modules/projectiles/guns/projectile/automatic.dm]: Convention is that guns should have a generic name. Specific name is put in the description.

**Comment by dragoslava** [comment]
https://forums.aurorastation.org/topic/11678-battle-rifle-amr-replacement/

dreamseeker_6JWLHf4TIA
dreamseeker_H27K2ldXuG

**Comment by ComradeCorbyn** [comment]
I still would like to see something changed with the colors. I can only barely see anything other then pitch black in it, and the ironsights look weird, as though it should be starting small from the end and getting to the apex at the end - if you catch my drift.

**Comment by dragoslava** [comment]
> I still would like to see something changed with the colors. I can only barely see anything other then pitch black in it, and the ironsights look weird, as though it should be starting small from the end and getting to the apex at the end - if you catch my drift.

will do

**Comment by alsoandanswer** [comment]
@dragoslava gave you a sprite on discord, give it a look

File: code/modules/projectiles/ammunition/boxes.dm [hunk 6]
```diff
@@ -245,6 +245,34 @@
...
+	mag_type = MAGAZINE
+	caliber = "A4.75x33"
+	ammo_type = /obj/item/ammo_casing/caseless475
+	multiple_sprites = 0
+	max_ammo = 45
+	icon = 'icons/obj/g11.dmi'
+
+
+/obj/item/ammo_magazine/g11/sniper
+	name = "battle rifle magazine(S Brand)"
```
Comment (skull132) [inline code/modules/projectiles/ammunition/boxes.dm]:
```suggestion
	name = "battle rifle magazine (S Brand)"
```

File: code/modules/projectiles/ammunition/bullets.dm [hunk 7]
```diff
@@ -42,6 +42,19 @@
 	caliber = "trod"
 	projectile_type = /obj/item/projectile/bullet/trod
 
+
+/obj/item/ammo_casing/caseless445m
+	desc = "caseless round."
```
Comment (skull132) [inline code/modules/projectiles/ammunition/bullets.dm]:
```suggestion
	desc = "caseless round"
```

File: code/modules/projectiles/ammunition/bullets.dm [hunk 8]
```diff
@@ -42,6 +42,19 @@
...
 
+
+/obj/item/ammo_casing/caseless445m
+	desc = "caseless round."
+	icon_state = "trod"
+	caliber = "445 caseless"
+	projectile_type = /obj/item/projectile/bullet/rifle/caseless445m
+
+/obj/item/ammo_casing/caseless475
+	desc = "caseless round."
```
Comment (skull132) [inline code/modules/projectiles/ammunition/bullets.dm]:
```suggestion
	desc = "caseless round"
```

**Pull request closed without merging**